### PR TITLE
recommendations on syntax and best practices of writing a TF module

### PR DESCRIPTION
**Pull request by givanovexpe**
* Using underscore in variable and resource names
* Renaming tags variable to beekeeper_tags for better visibility
* Renaming some variables to keep consistent naming
* Splitting the docker image and version into separate vars for visibility
* Adding instance_name variable to differentiate between multiple Beekeeper installations (multi-region)
* Limiting S3 access to Beekeeper IAM role to a list of allowed buckets
* Adding more RDS parameters - backups, maintenance window
* Creating RDS subnet group in the module
* Splitting the IAM roles and policies into separate files
* Adding cloudwatch dashboard and alerts for the Beekeeper ECS service
* Creating Beekeeper security groups in the module
* Deploying the lambda slack notifier into the VPC (better security)
* Add `docker_registry_auth_secret_name` variable to fetch from AWS Secrets Manager a Docker Registry Auth if the registry is protected

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+## 3. Add unit tests
+
+If your contribution modifies existing or adds new code please add corresponding unit tests for this.
+
+## 4. Ensure that the build passes
+
+Run
+```shell
+mvn package
```
Comment (massdosage) [inline CONTRIBUTING.md]: This isn't a mvn project so you should either remove this step or replace it with a Terraform equivalent.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+## 2. One feature, one branch
+
+Work for each new feature/issue should occur in its own branch. To create a new branch from the command line:
+```shell
+git checkout -b my-new-feature
+```
+where "my-new-feature" describes what you're working on.
+
+## 3. Add unit tests
```
Comment (massdosage) [inline CONTRIBUTING.md]: Is this even possible for terraform?

File: lambda/cloudwatch.tf [hunk 3]
```diff
@@ -1,5 +1,11 @@
-resource "aws_cloudwatch_metric_alarm" "beekeeper-sqs-dlq" {
-  alarm_name          = "${var.queue-name}-dlq-alarm"
+/**
+ * Copyright (C) 2018-2019 Expedia Inc.
```
Comment (massdosage) [inline lambda/cloudwatch.tf]: Needs to be `Expedia, Inc`

File: lambda/common.tf [hunk 4]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Copyright (C) 2018-2019 Expedia Inc.
```
Comment (massdosage) [inline lambda/common.tf]: The comma, as above and for all the other headers.

File: main/iam-policy-secretsmanager.tf [hunk 5]
```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (C) 2018-2019 Expedia Inc.
```
Comment (massdosage) [inline main/iam-policy-secretsmanager.tf]: Why do some license files say 2018-2019 and same say 2019?

File: CONTRIBUTING.md (hunk 1 above)
Comment (max-jacobs) [reply CONTRIBUTING.md]: will exchange for `terraform plan`

File: CONTRIBUTING.md (hunk 2 above)
Comment (max-jacobs) [reply CONTRIBUTING.md]: probably not... will remove

File: main/iam-policy-secretsmanager.tf (hunk 5 above)
Comment (max-jacobs) [reply main/iam-policy-secretsmanager.tf]: fixed

**Pull request merged**